docs/api_requests/block-vhost-user.md

```diff
@@ -0,0 +1,189 @@
+# Vhost-user block device
+
+As an alternative to [block device](block-io-engine.md) `Sync` and `Async`
+engines, Firecracker supports a vhost-user block device.
+
+There is a good introduction of how a vhost-user block device works
+in general at [FOSDEM23](https://archive.fosdem.org/2023/schedule/event/sds_vhost_user_blk).
+
+[Vhost-user](https://qemu-project.gitlab.io/qemu/interop/vhost-user.html)
+is a userspace protocol that allows to delegate Virtio queue processing
+to another userspace process on the host, as opposed to performing this task
+within Firecracker's VMM thread.
+
+In the vhost-user architecture, the VMM acts as a vhost-user frontend, and
+it is responsible for connecting to the backend via a Unix domain socket (UDS)
+and sharing sufficient information about the guest memory and Virtio queues.
+The vhost-user backend is supposed to receive all this information and perform
+handling of guests's requests. The UDS socket is only used for control purposes
+and does not participate in the dataplane.
+
+Firecracker only implements a vhost-user frontend. Users are free
+to choose from [existing open source backends](#backends) or implement their
+own.
+
+## Topology
+
+Each vhost-user device connects to its own UDS socket. There is no way for
+multiple devices to share a single socket, as there is no way to differentiate
+messages related to devices.
+
+Each device can be served by a separate backend or a single backend can serve
+multiple devices.
+
+## Interactions with the backend
+
+There are two points when the vhost-user frontend communicates with
+the backend:
+
+1. Device initialisation. When a vhost-user device is created, Firecracker
+   connects to the corresponding UDS socket and negotiates Virtio and Vhost
+   features with backend and retrieves device configuration.
+1. Device activation. When the guest driver finishes setting up the device,
+   Firecracker shares memory tables and Virtio queue information with the backend.
+   As a part of this, Firecracker shares file descriptors for guest's memory
+   regions, as well as file descriptors for queue notifications.
```

Comment:
Just a thought (not necessarily actionable in this PR). A lot of this info is `vhost-user` generic, rather than `vhost-user-block` specific. Do you see value in having a separate doc for `vhost-user` generic info?

Reply:
Thoughts about this? If you don't think we should do this here feel free to resolve.

Reply:
Works for me! I was mainly asking your view if that makes sense?